Add requirements section. (#8)
NLnetLabs/incremental-deleg: draft-homburg-deleg-incremental-deleg.md

Information about which authoritative name servers and with capabilities, MAY be provided in a IDELEG RRset directly at the `_deleg` label under the apex of the zone.
Priming queries from a incremental deleg supporting resolver, MUST send an `_deleg.<apex> IDELEG` query in parallel to the legacy `<apex> NS` query and process the content as if it was found through an incremental referral response.

# How does incremental deleg meet the requirements

> This section will discuss how incremental deleg meets the requirements for a new delegation mechanism as decribed in {{?I-D.ietf-deleg-requirements-02}}

Reviewer: ```suggestion
This section will discuss how incremental deleg meets the requirements for a new delegation mechanism as described in {{?I-D.ietf-deleg-requirements-02}}
```

+ H1. DELEG must not disrupt the existing registration model of domains.

  The existing zone structure including the concept of delegations from
  a parent zone to a child zone is left unchanged.

+ H2. DELEG must be backwards compatible with the existing ecosystem.

  The new delegations do not interfere with legacy software.

  The behavior of incremental deleg-aware resolvers includes a fallback to NS
  records if no incremental delegation is present (See {{recursive-resolver-behavior}}).

+ H3. DELEG must not negatively impact most DNS software.

>   Incremental deleg introduces a new Rtype.

Reviewer: ```suggestion
  Incremental deleg introduces a new RR type.
```

  Software that parses zone file format needs to be changed to support the new
  type.

>   Though unknown type notation ({{!RFC3597}}) can be used in some cases if no support for the new Rtype is present.

Reviewer: ```suggestion
  Though unknown type notation {{!RFC3597}} can be used in some cases if no support for the new RR type is present.
```
The renderer already adds square brackets

>   Existing authoritatives can serve incremental deleg-aware zones (though less efficiently), existing signers can sign incremental deleg-aware zones, existing diagnostic tools can query incremental deleg-aware zones.

Reviewer: ```suggestion
  Existing authoritatives can serve incremental deleg zones (though less efficiently), existing signers can sign incremental deleg zones, existing diagnostic tools can query incremental deleg zones.
```
Remove the "-aware" because it may make one feel that there is some "awareness" needed in the implementation.

  Non-recursive DNSSEC validators can operate independently from (possibly legacy) recursive resolvers.

+ H4. DELEG must be able to secure delegations with DNSSEC.

  Incremental delegations are automatically secured with DNSSEC
  (if the parent zone is signed). A replacement for DS records is described in {{?I-D.homburg-deleg-incremental-dnssec}}.

+ H5. DELEG must support updates to delegation information with the same relative ease as currently exists with NS records.

  Incremental delegations are affected by TTL like any other
  DNS record.

+ H6. DELEG must be incrementally deployable and not require any sort of flag day of universal change.

  Incremental deleg-aware zones can be added without upgrading authoritatives.

>   Incremental deleg-aware zones still work with old resolvers and validators.

Reviewer: ```suggestion
  Incremental deleg zones can be added without upgrading authoritatives.
  Incremental deleg zones still work with old resolvers and validators.
```

  Basically any combination of old and new should work, though with
  reduced efficiency for some combinations.

+ H7. DELEG must allow multiple independent operators to simultaneously serve a zone.

  Incremental deleg allows for multiple IDELEG records. This allows
  multiple operators to serve the zone.

+ S1. DELEG should facilitate the use of new DNS transport mechanisms

  New transports are already defined for the DNS mode of SVCB ({{!RFC9461}}).
  The same parameters are used for IDELEG.

+ S2. DELEG should make clear all of the necessary details for contacting a service

  Most of the needed SVCB parameters are already defined in existing standards.

>   The exception is a replacemenet for the DS records, which is described in {{?I-D.homburg-deleg-incremental-dnssec}}.

Reviewer: ```suggestion
  The exception is a replacement for the DS records, which is described in {{?I-D.homburg-deleg-incremental-dnssec}}.
```
